src/Bicep.Core/Semantics/Namespaces/NamespaceResolver.cs

```diff
@@ -37,7 +38,8 @@ void TryAddBuiltInNamespace(string @namespace)
                     return;
                 }
 
-                if (namespaceProvider.TryGetNamespace(@namespace, @namespace, targetScope) is not { } namespaceType)
+                // This check is only validating that the namespace exists (and not which aspects of it are enabled), so it's OK to use a bogus FeatureProvider
```

Comment:
I'd be concerned that this expectation might change in future. Is it going to be complex to thread the real featureprovider through?